graphql_compiler/schema_merging/rename_schema.py

```diff
@@ -0,0 +1,232 @@
+# Copyright 2019-present Kensho Technologies, LLC.
+from collections import namedtuple
+
+from graphql import build_ast_schema, parse
+from graphql.language.visitor import Visitor, visit
+
+from .utils import SchemaError, get_schema_data
+
+
+RenamedSchema = namedtuple(
+    'RenamedSchema', [
+        'schema_ast',  # type: Document, ast representing the renamed schema
+        'reverse_name_map',  # type: Dict[str, str], new type name to old type name
+        'reverse_root_field_map'  # type: Dict[str, str], new root field name to old root field
+    ]
+)
+
+
+def rename_schema(schema_string, rename_func=lambda name: name):
```

Comment:
What's the advantage of taking a lambda over a dict here?

Reply:
It felt to me that using a function instead of a dict to transform names is more convenient and clear in some ways, but I'm not completely sure which is better. If we use a dict, then we could take one of two approaches. 
One, the dict is required to contain all type names and field names. This would require the user to get all types of a schema, which can be inconvenient.
Two, the dict contains some subset of types and fields, and all other types and fields pass through without change. I feel like this approach could lead to errors, but I'm not completely sure.

In particular, I imagined this being used something like a namespace, by prefixing a string to all names of a schema. In that case, a func seems much more convenient than a dict.
If the user only wants to rename a few conflicting names, they can still define their rename_func to be something like: if input string is a key in this dict, return the value associated with that key, otherwise return the original string.